### ISSUE TITLE
Spacer at the edge of a panel isn't preserved after reboot

### ISSUE THREAD

**Issue by iron3oxide** [BuddiesOfBudgie/budgie-desktop#72]
## Description
I added a spacer to the right edge of my top panel so that the widget wouldn't be squished into the corner so much. While I am able to manually accomplish the desired behavior with the Budgie Desktop Settings, the spacer isn't shown anymore after a reboot. 

## Budgie version
budgie-desktop 10.5.3 (git-26cde8fd300a87aba0d085ca50c89728150138d2)

## Operating system
Ultramarine Linux 21.12 Revision 1

## Steps to reproduce the issue
1. add a spacer to the right (or probably any edge) of your panel
2. reboot

## Actual result
the spacer isn't shown in the panel

## Expected result
some space between the edge of the screen and the rightmost widget

## Additional information
I have a spacer left to the rightmost widget as well to make it stand out more.

**Comment by AppleofPower** [comment]
This also happens in Solus Linux on budgie 10.6.

**Comment by serebit** [comment]
We're having trouble reproducing this on both EndeavourOS and Solus. Can either of you send a screenshot of your panel configuration so we can try to better approximate your setup?

**Comment by fossfreedom** [comment]
I suspect this is the old "wandering icon" issue - where applets keep shifting their position between session restarts.

Remember this issue is reproducible if a distro has a panel.ini / layouts with predefined applets - and one or more of those predefined applets is uninstalled.  Also can occasionally reproduce by having a thirdparty applet on the panel and uninstalling that applet without first removing the applet from the panel.

Does need resolving ... and the only workaround is to ensure all panel.ini / layouts applets are reinstalled before doing a panel reset.

**Comment by iron3oxide** [comment]
Panel settings:
![Screenshot from 2022-03-19 21-50-25](https://user-images.githubusercontent.com/98779754/159138052-eaad2d8c-678f-4e38-9613-14b18f6378f5.png)

Desired state, achieved only after shuffling the items around a bit:
![Screenshot from 2022-03-19 21-51-29](https://user-images.githubusercontent.com/98779754/159138080-6e15bff6-2b7c-4048-a9a8-569819cb2760.png)

Actual state after reboot:
![Screenshot from 2022-03-19 21-49-37](https://user-images.githubusercontent.com/98779754/159138093-4efda8d0-b8a7-4149-ae4b-935e793f7eaa.png)

**Comment by serebit** [comment]
Yeah, looks like this could be an issue with third-party applets as @fossfreedom suggested. I'll mark this as verified, but I will warn that this issue is pretty fundamental to how the panel works, and may take a while to fix.

**Comment by fossfreedom** [comment]
there probably many parts of this.  After putting some debug code into the settings what I found for the following scenario is a mismatch between the panel applet "position" number within a region and the actual number of applets in a specific region.  What I mean by region is the start, center and end boxes. What I mean by position number is look at the dconf com.solus-project.budgie-panel.applets.{xyz}.position

So I uninstalled an applet that exists on the end panel region without removing the applet first from the panel.

On login, the applets in the panel will have their same position number as before - so 0, 1, 3, 4, 5 - i.e. applet that was at position 2 is now removed.

In the settings code its check for when moving an applet down the list is "is the position >= the number of applets in a region" - so the down arrow will become disabled towards the end of the region when  the last but one is clicked - the position number of the applet is > than the children (number of applets in the end region).

If you move all the applets from one region to another ... then back again the position number will be correct again - so no wandering icon.

How best to resolve this specific scenario I am not sure.  Guessing maybe when the applets are being added to a region - if very first applet being added to a region - its position does not start at zero - or the position number jumps compared to the previous then the position number should be recalculated.
